styx-common/src/main/java/com/spotify/styx/model/data/Execution.java

```diff
@@ -41,19 +42,44 @@
   @JsonProperty
   public abstract Optional<String> commitSha();
 
+  @JsonProperty
+  public abstract Optional<FlyteExecConf> flyteExecConf();
+
   @JsonProperty
   public abstract Optional<String> runnerId();
 
   @JsonProperty
   public abstract List<ExecStatus> statuses();
 
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
   @JsonCreator
   public static Execution create(
       @JsonProperty("execution_id") Optional<String> executionId,
       @JsonProperty("docker_image") Optional<String> dockerImage,
       @JsonProperty("commit_sha") Optional<String> commitSha,
+      @JsonProperty("flyte_exec_conf") Optional<FlyteExecConf> flyteExecConf,
       @JsonProperty("runner_id") Optional<String> runnerId,
       @JsonProperty("statuses") List<ExecStatus> statuses) {
-    return new AutoValue_Execution(executionId, dockerImage, commitSha, runnerId, statuses);
+    return new AutoValue_Execution(executionId, dockerImage, commitSha, flyteExecConf, runnerId, statuses);
```

Comment:
Commenting on myself. I will add a check to make sure that `dockerImage` and `flyteExecConf` both cannot be present at the same time